staging/src/k8s.io/api/resource/v1beta2/types.go

```diff
@@ -0,0 +1,1560 @@
+// in a ResourceSlice. This is used to validate the fields:
+// * spec.sharedCounters
+const ResourceSliceMaxSharedCounters = 32
+
+// Defines the max number of Counters from which a device
+// can consume. This is used to validate the fields:
+// * spec.devices[].consumesCounter
+const ResourceSliceMaxDeviceCounterConsumptions = 32
+
+// Defines the max number of counters
+// that can be specified for sharedCounters in a ResourceSlice.
+// This is used to validate the fields:
+// * spec.sharedCounters[].counters
+const ResourceSliceMaxSharedCountersCounters = 32
+
+// Defines the max number of counters
+// that can be specified for consumesCounter in a ResourceSlice.
+// This is used to validate the fields:
+// * spec.devices[].consumesCounter[].counters
+const ResourceSliceMaxDeviceCounterConsumptionCounters = 32
+
+// Device represents one individual hardware instance that can be selected based
+// on its attributes. Besides the name, exactly one field must be set.
+type Device struct {
+	// Name is unique identifier among all devices managed by
+	// the driver in the pool. It must be a DNS label.
+	//
+	// +required
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
+
+	// Attributes defines the set of attributes for this device.
+	// The name of each attribute must be unique in that set.
+	//
+	// The maximum number of attributes and capacities combined is 32.
+	//
+	// +optional
+	Attributes map[QualifiedName]DeviceAttribute `json:"attributes,omitempty" protobuf:"bytes,2,rep,name=attributes"`
+
+	// Capacity defines the set of capacities for this device.
+	// The name of each capacity must be unique in that set.
+	//
+	// The maximum number of attributes and capacities combined is 32.
+	//
+	// +optional
+	Capacity map[QualifiedName]DeviceCapacity `json:"capacity,omitempty" protobuf:"bytes,3,rep,name=capacity"`
+
+	// ConsumesCounter defines a list of references to sharedCounters
+	// and the set of counters that the device will
+	// consume from those counter sets.
+	//
+	// There can only be a single entry per counterSet.
+	//
+	// The maximum number of device counter consumption entries
+	// is 32. This is the same as the maximum number of shared counters
+	// allowed in a ResourceSlice.
+	//
+	// +optional
+	// +listType=atomic
+	// +featureGate=DRAPartitionableDevices
+	ConsumesCounter []DeviceCounterConsumption `json:"consumesCounter,omitempty" protobuf:"bytes,4,rep,name=consumesCounter"`
+
+	// NodeName identifies the node where the device is available.
+	//
+	// Must only be set if Spec.PerDeviceNodeSelection is set to true.
+	// At most one of NodeName, NodeSelector and AllNodes can be set.
+	//
+	// +optional
+	// +oneOf=DeviceNodeSelection
+	// +featureGate=DRAPartitionableDevices
+	NodeName *string `json:"nodeName,omitempty" protobuf:"bytes,5,opt,name=nodeName"`
+
+	// NodeSelector defines the nodes where the device is available.
+	//
+	// Must use exactly one term.
```

Comment:
The corresponding field in v1beta1 lacks this comment. Can you add it there?

Reply:
Fixed